frontend/src/views/admin/assignments/offer-actions.js

```diff
@@ -165,53 +172,85 @@ function ConfirmWithDialogActionButton(props) {
             selectedAssignments.every(OfferTest[key]);
     }
 
-    const actionConfirmation = {
-        accept: confirmOfferAccept,
-        create: confirmOfferCreate,
-        nag: confirmOfferNag,
-        reject: confirmOfferReject,
-        withdraw: confirmOfferWithdraw,
-        email: confirmOfferEmail,
-    };
-
-    const actionCallback = {
-        accept: acceptOffers,
-        create: createOffers,
-        nag: nagOffers,
-        reject: rejectOffers,
-        withdraw: withdrawOffers,
-        email: emailOffers,
-    };
-
-    const dialogVisible = {
-        accept: showAcceptConfirmation,
-        create: showCreateConfirmation,
-        nag: showNagConfirmation,
-        reject: showRejectConfirmation,
-        withdraw: showWithdrawConfirmation,
-        email: showEmailConfirmation,
-    };
-
-    const dialogSetVisible = {
-        accept: setShowAcceptConfirmation,
-        create: setShowCreateConfirmation,
-        nag: setShowNagConfirmation,
-        reject: setShowRejectConfirmation,
-        withdraw: setShowWithdrawConfirmation,
-        email: setShowEmailConfirmation,
-    };
-
     return (
         <React.Fragment>
-            <ActionButtons
-                actionPermitted={actionPermitted}
-                actionConfirmation={actionConfirmation}
+            <ConfirmWithDialogActionButton
+                buttonIcon={<FaUserPlus />}
+                actionConfirmation={confirmOfferCreate}
+                buttonEnable={actionPermitted.canCreate}
+                buttonText="Create Offer"
+                selectedAssignments={selectedAssignments}
+                dialogVisible={showCreateConfirmation}
+                dialogSetVisible={setShowCreateConfirmation}
+                actionCallback={createOffers}
+                dialogTitle="Creating Multiple Offers"
+                dialogBody={`You are creating the following ${selectedAssignments.length} offers:`}
+                dialogConfirmation={`Create ${selectedAssignments.length} Offers`}
+            />
+            <ConfirmWithDialogActionButton
+                buttonIcon={<FaUserTimes />}
+                actionConfirmation={confirmOfferWithdraw}
+                buttonEnable={actionPermitted.canWithdraw}
+                buttonText="Withdraw Offer"
+                selectedAssignments={selectedAssignments}
+                dialogVisible={showWithdrawConfirmation}
+                dialogSetVisible={setShowWithdrawConfirmation}
+                actionCallback={withdrawOffers}
+                dialogTitle="Withdrawing Multiple Offers"
+                dialogBody={`You are withdrawing from the following ${selectedAssignments.length} offers:`}
+                dialogConfirmation={`Withdraw ${selectedAssignments.length} Offers`}
+            />
```

Comment:
We're getting closer, but this is still not right. I didn't mean for you to invent a new abstract `ConfirmWithDialogAction` component. I meant for you to create 7 concrete components which repeat the needed code from `ConfirmWithDialogAction`.